docs/csharp/language-reference/builtin-types/struct.md

```diff
@@ -96,9 +96,32 @@ In the case of the [built-in value types](value-types.md#built-in-value-types),
 
 When you pass a structure-type variable to a method as an argument or return a structure-type value from a method, the whole instance of a structure type is copied. That can affect the performance of your code in high-performance scenarios that involve large structure types. You can avoid value copying by passing a structure-type variable by reference. Use the [`ref`](../keywords/ref.md#passing-an-argument-by-reference), [`out`](../keywords/out-parameter-modifier.md), or [`in`](../keywords/in-parameter-modifier.md) method parameter modifiers to indicate that an argument must be passed by reference. Use [ref returns](../../programming-guide/classes-and-structs/ref-returns.md) to return a method result by reference. For more information, see [Write safe and efficient C# code](../../write-safe-efficient-code.md).
 
+## `ref` struct
+
+Beginning with C# 7.2, you can use the `ref` modifier in the declaration of a structure type. Instances of a `ref` struct type are allocated on the stack and can't escape to the managed heap. To ensure that, the compiler limits the usage of `ref` struct types as follows:
+
+- A `ref` struct can't be the element type of an array.
+- A `ref` struct can't be a declared type of a field of a class or a non-`ref` struct.
+- A `ref` struct can't implement interfaces.
+- A `ref` struct can't be boxed to <xref:System.ValueType?displayProperty=nameWithType> or <xref:System.Object?displayProperty=nameWithType>.
+- A `ref` struct can't be a type argument.
+- A `ref` struct variable can't be captured by a [lambda expression](../../programming-guide/statements-expressions-operators/lambda-expressions.md) or a [local function](../../programming-guide/classes-and-structs/local-functions.md).
+- A `ref` struct variable can't be used in an [`async`](../keywords/async.md) method. However, you can use `ref` struct variables in synchronous methods, for example, in those that return <xref:System.Threading.Tasks.Task> or <xref:System.Threading.Tasks.Task%601>.
+- A `ref` struct variable can't be used in [iterators](../../iterators.md).
+
+In particular, you use a `ref` struct type if you want to encapsulate other `ref` struct types:
```

Comment:
This is a bit misleading. It's true, but `ref struct` types can also contain unmanaged types (int, short, etc).

Reply:
I'll swap the sentence. Its purpose is to introduce one possible use case of `ref` structs.

By the way, `ref` structs can also contain fields of reference types

Reply:
@BillWagner I've rephrased that sentence completely. Please check the latest commit.